src/propstepsolvers.jl

```diff
@@ -34,48 +34,48 @@ QuKrylov() = QuKrylov(@compat Dict(:NC=>10))
 Propagates to the next time state
 Input Parameters:
 `prob`             :  Method to be used
-`op`               :  Hamiltonian of the system
+`eq`               :  Quantum Equation type
 `t`                :  Time corresponding to the t_state
 `current_t`        :  Current time
 `current_qustate`  :  Quantum state corresponding to current time
 """ ->
-function propagate(prob::QuEuler, op, t, current_t, current_qustate)
+function propagate(prob::QuEuler, eq::QuEquation, t, current_t, current_qustate)
     dt = t - current_t
-    return (eye(op)-im*op*dt)*current_qustate
+    return (eye(operator(eq))-im*operator(eq)*dt)*vec(current_qustate)
 end
 
-function propagate(prob::QuCrankNicolson, op, t, current_t, current_qustate)
+function propagate(prob::QuCrankNicolson, eq::QuEquation, t, current_t, current_qustate)
     dt = t - current_t
-    uni = eye(op)-im*op*dt/2
-    return \(uni', uni*current_qustate)
+    uni = eye(operator(eq))-im*operator(eq)*dt/2
+    return \(uni', uni*vec(current_qustate))
 end
 
-function propagate(prob::QuKrylov, op, t, current_t, current_qustate)
+function propagate(prob::QuKrylov, eq::QuEquation, t, current_t, current_qustate)
     dt = t - current_t
-    basis_size = get(prob.options,:NC, length(current_qustate))
-    N = min(basis_size, length(current_qustate))
-    v = Array(typeof(current_qustate),0)
+    basis_size = get(prob.options,:NC, length(vec(current_qustate)))
```

Comment:
I think it might be cleaner to define `psi=vec(current_state)` at the beginning of the function and then only use that.